message_ix/models.py

```diff
@@ -415,6 +419,61 @@ def enforce(scenario):
                 scenario.remove_set(set_name, existing)
                 scenario.add_set(set_name, expected)
 
+        # Enforcing new indexes for existing set and parameters
```

Comment:
@khaeru, following the discussion under #631, I drafted a workflow to help reinitialize some existing sets and parameters here. The code checks if the item is empty before re-initializing, and if not raises an exception. I do believe this can be done much more compact and maybe under a separate method than `enforce()`.

Reply:
@khaeru Resuming working on this, I was wondering if you have time to have a look at the above point. In principle, it's a design issue; how we would like to alert users that they need to re-parameterize some existing sets and parameters in their scenario to make it work with the enhanced code.

Reply:
@behnam-zakeri I've just pushed a couple commits with the requested changes.

Per this:
> users that they need to re-parameterize some existing sets and parameters in their scenario to make it work with the enhanced code.

Did you already write some documentation or helper code about _how_ they should do this? I think we ought to provide some…

Reply:
I pushed another couple of commits to add a helper function `expand_dims()`.

Reply:
Thanks a lot, @khaeru both for expanding the doc, the elegant workaround for adding the re-initialized items, and the help function. It looks like very well-structured code and fits the purpose nicely.